tests/psa-client-server/psasim/tools/psa_autogen.py

```diff
@@ -19,6 +19,9 @@
 GENERATED_H_PATH = os.path.join(SCRIPT_PATH, "..", "include", "psa_manifest")
 GENERATED_C_PATH = os.path.join(SCRIPT_PATH, "..", "src")
 
+def open_in_include_dir(filename: str):
+    return open(os.path.join(GENERATED_H_PATH, filename), "w")
```

Comment:
I am not a fan of this syntax, since it will not close the file handler. I would remove this function and use nested with statements instead,

with open(os.path.join(GENERATED_H_PATH, "pid.h"), "w"):
    with open(os.path.join(GENERATED_H_PATH, "sid.h"), "w"):
         with open(os.path.join(GENERATED_H_PATH, FILENAME + ".h"), "w"): 
             ...other logic....


Alternatively we can use striongIO objects to compile our files and then write them into the headers.

Reply:
Ok, but those files are opened inside another `with open()` block (for reading the input file) and they are written to in several lines along the script. Following this approach means that a good part of the script would be right indented a lot. Don't get me wrong, I agree with you that what you are proposing is the right way of writing to a file in a python script, but I'm only stating that this change basically requires/suggests to heavily modify that script.

Reply:
We can keep the `open_in_include_dir` function, but it should be used like `open`, i.e.
```
with open_in_include_dir('foo.h') as foo:
    ... foo.read() ...
```

> a good part of the script would be right indented a lot

That's a hint to break the code into more auxiliary functions/methods.